test/propshaft/compilers/css_asset_urls_test.rb

```diff
@@ -0,0 +1,110 @@
+require "test_helper"
+require "minitest/mock"
+require "propshaft/asset"
+require "propshaft/assembly"
+require "propshaft/compilers"
+
+class Propshaft::Compilers::CssAssetUrlsTest < ActiveSupport::TestCase
+  setup do
+    @assembly = Propshaft::Assembly.new(ActiveSupport::OrderedOptions.new.tap { |config| 
+      config.paths = [ Pathname.new("#{__dir__}/../../fixtures/assets/vendor") ]
+      config.output_path = Pathname.new("#{__dir__}/../../fixtures/output")
+      config.prefix = "/assets"
+    })
+
+    @assembly.compilers.register "text/css", Propshaft::Compilers::CssAssetUrls
+  end
+
+  test "basic" do
+    compiled = compile_asset_with_content(%({ background: url(file.jpg); }))
+    assert_match /{ background: url\("\/assets\/foobar\/source\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "blank spaces around name" do
+    compiled = compile_asset_with_content(%({ background: url( file.jpg ); }))
+    assert_match /{ background: url\("\/assets\/foobar\/source\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "quotes around name" do
+    compiled = compile_asset_with_content(%({ background: url("file.jpg"); }))
+    assert_match /{ background: url\("\/assets\/foobar\/source\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "single quotes around name" do
+    compiled = compile_asset_with_content(%({ background: url('file.jpg'); }))
+    assert_match /{ background: url\("\/assets\/foobar\/source\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "root directory" do
+    compiled = compile_asset_with_content(%({ background: url('/file.jpg'); }))
+    assert_match /{ background: url\("\/assets\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "same directory" do
+    compiled = compile_asset_with_content(%({ background: url('./file.jpg'); }))
+    assert_match /{ background: url\("\/assets\/foobar\/source\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "subdirectory" do
+    compiled = compile_asset_with_content(%({ background: url('./images/file.jpg'); }))
+    assert_match /{ background: url\("\/assets\/foobar\/source\/images\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "parent directory" do
+    compiled = compile_asset_with_content(%({ background: url('../file.jpg'); }))
+    assert_match /{ background: url\("\/assets\/foobar\/file-[a-z0-9]{40}.jpg"\); }/, compiled
+  end
+
+  test "grandparent directory" do
+    compiled = compile_asset_with_content(%({ background: url('../../file.jpg'); }))
```

Comment:
Broken. Capture group is only returning a single occurence of `../` in `match[1]`